store/CSDK/test/build_carbondata_test.sh

```diff
@@ -0,0 +1,52 @@
+
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+export CFLAGS=`pkg-config --cflags thrift`" -g3 -O0"
+export LDFLAGS=`pkg-config --libs-only-L thrift`
+export LIBS=`pkg-config --libs-only-l thrift`
+
+export FORMAT_FILE_PATH=../../../format/src/main/thrift
+
+thrift --gen cpp -o . $FORMAT_FILE_PATH/carbondata_index_merge.thrift 
+thrift --gen cpp -o . $FORMAT_FILE_PATH/carbondata_index.thrift
+thrift --gen cpp -o . $FORMAT_FILE_PATH/carbondata.thrift
+thrift --gen cpp -o . $FORMAT_FILE_PATH/dictionary_metadata.thrift
+thrift --gen cpp -o . $FORMAT_FILE_PATH/dictionary.thrift
+thrift --gen cpp -o . $FORMAT_FILE_PATH/schema.thrift
+thrift --gen cpp -o . $FORMAT_FILE_PATH/sort_index.thrift
+
+g++ gen-cpp/carbondata_constants.cpp -c $CFLAGS
+g++ gen-cpp/carbondata_index_constants.cpp -c $CFLAGS
+g++ gen-cpp/carbondata_index_merge_constants.cpp -c $CFLAGS
+g++ gen-cpp/carbondata_index_merge_types.cpp -c $CFLAGS
+g++ gen-cpp/carbondata_index_types.cpp -c $CFLAGS
+g++ gen-cpp/carbondata_types.cpp -c $CFLAGS
+g++ gen-cpp/dictionary_constants.cpp -c $CFLAGS
+g++ gen-cpp/dictionary_metadata_constants.cpp -c $CFLAGS
+g++ gen-cpp/dictionary_metadata_types.cpp -c $CFLAGS
+g++ gen-cpp/dictionary_types.cpp -c $CFLAGS
+g++ gen-cpp/schema_constants.cpp -c $CFLAGS
+g++ gen-cpp/schema_types.cpp -c $CFLAGS
+g++ gen-cpp/sort_index_constants.cpp -c $CFLAGS
+g++ gen-cpp/sort_index_types.cpp -c $CFLAGS
+
+g++ carbondata_test.cpp -c $CFLAGS -I./gen-cpp/
+
+g++ carbondata_constants.o carbondata_index_constants.o carbondata_index_merge_constants.o \
```

Comment:
Can you move the generated files to other folder? like target,  because git status will show the result files.:
```

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	store/CSDK/test/carbondata_constants.o
	store/CSDK/test/carbondata_index_constants.o
	store/CSDK/test/carbondata_index_merge_constants.o
	store/CSDK/test/carbondata_index_merge_types.o
	store/CSDK/test/carbondata_index_types.o
	store/CSDK/test/carbondata_test
	store/CSDK/test/carbondata_test.o
	store/CSDK/test/carbondata_types.o
	store/CSDK/test/dictionary_constants.o
	store/CSDK/test/dictionary_metadata_constants.o
	store/CSDK/test/dictionary_metadata_types.o
	store/CSDK/test/dictionary_types.o
	store/CSDK/test/gen-cpp/
	store/CSDK/test/schema_constants.o
	store/CSDK/test/schema_types.o
	store/CSDK/test/sort_index_constants.o
	store/CSDK/test/sort_index_types.o
```

Reply:
move to 'store/CSDK/target'